File: app/assets/javascripts/models/codes/KeCode.js
define([
    'models/abstract/CategorisedDataset'
], function(
    CategorisedDataset
) {

    function KeCode() {
        var _this = new CategorisedDataset();

        /**
         * @returns {string}
         */
        _this.name = function() {
            return 'Ke Code Dataset';
        };

        return _this;
    }

    KeCode.ID = 'ke';
    return KeCode;
});

File: app/models/codes/ke_code.rb
require 'abstract/categorised_code'
require 'catalog'

class KeCode < CategorisedCode
  pragmatize!

  @name ='KeCode'
  @id = :ke
  @type_description = 'Ke code means something and does something'
end

# Adds an extension method to code catalog
# annotated with _code pragma
# to indicate that chop code exists
class Catalog
  pragmatize!

  _code
  def ke_code_in(context)
    KeCode
  end
end